### PR TITLE
Make yadism compatible with eko 0.11

### PR DESCRIPTION
**Pull request by andreab1997**
At the end, for the `convolute_operator` function, it seems that changing `xgrid_raw` to `xgrid.raw` was enough. Anyway I have opened this PR in case I will find other things to be changed in the next future. @AleCandido @felixhekhorn

### CODE REVIEW

**Review by AleCandido — APPROVED** [review]
The change is minimal and needed. I didn't run myself, but I agree that this change has to be done

**Review by AleCandido — CHANGES_REQUESTED** [review]
Sorry if I change my mind (slightly).
Actually, it is currently broken, since you kept:
https://github.com/NNPDF/yadism/blob/a7dea9a2c072dd82780b1238e7a59f218dda85a1/pyproject.toml#L43

Update it accordingly, and run `poetry update`. Then you can merge :)

File: tests/yadism/test_tmc.py
Comment (giacomomagni) [inline tests/yadism/test_tmc.py:162]: I have no clue why I had to change this in order to pass...

Comment (AleCandido) [reply tests/yadism/test_tmc.py:162]: If you want, you can add `rtol` also using `pytest.approx`. We are using `np.testing`  mainly when arrays are involved, but [`pytest.approx`](https://docs.pytest.org/en/7.1.x/reference/reference.html#pytest-approx) actually works also in that case (so it is a complete replacement)

However, yes, it is weird...

File: tests/yadism/test_output.py
Comment (AleCandido) [inline tests/yadism/test_output.py:29]: Now there are two `xgrid`s (`out["xgrid]` and `out["interpolation_xgrid"]`) keep a single one

File: tests/yadism/test_tmc.py
Comment (AleCandido) [inline tests/yadism/test_tmc.py:165]: And this is weird as well (but essentially the same of above...)

**Review by AleCandido — CHANGES_REQUESTED** [review]

**Comment by AleCandido** [comment]
At this point I'd even approve the PR, but if possible it would be interesting to know more about the precision in the TMC integration.

In principle, it has not been touched (just two characters), so nothing should changed...
And an increment in tolerance can mask an actual bug. Since only the xgrid has been changed, I'd point to something like wrong usage of xgrid and its logarithm (something like `xgrid.raw` vs `xgrid.grid`), but in principle almost nothing has changed, so it is still weird...

**Comment by giacomomagni** [comment]
> In principle, it has not been touched (just two characters), so nothing should changed... And an increment in tolerance can mask an actual bug. Since only the xgrid has been changed, I'd point to something like wrong usage of xgrid and its logarithm (something like `xgrid.raw` vs `xgrid.grid`), but in principle almost nothing has changed, so it is still weird...

I fear I'm not able to understand the origin of this... @andreab1997 might have a clue?
 I tried a benchmark with apfel:
```
tmc = BenchmarkTMC()
tmc.benchmark_nlo()
```

and indeed it's not working.

**Comment by AleCandido** [comment]
If even benchmarks are broken, I suspect we are introducing a bug, so I'd block merging for as long as we do not understand what is going on. @felixhekhorn

**Comment by giacomomagni** [comment]
> If even benchmarks are broken, I suspect we are introducing a bug, so I'd block merging for as long as we do not understand what is going on. @felixhekhorn

Sorry, I noticed only now that there was a comment about FL with TMC being broken in apfel, so I'd amend my statement, so `F2_total` and `F3_total` with `FONLL-A` and `TMC` don't display any major difference.

File: tests/yadism/test_tmc.py
Comment (felixhekhorn) [reply tests/yadism/test_tmc.py:162]: Might it be that we change from linear interpolation to log interpolation? because now https://github.com/NNPDF/eko/blob/e3466fae35fd245c7e1a99158d8582f590276555/src/eko/interpolation.py#L501-L505 by default we're  log ... before there was the explict `log=False` argument (see https://github.com/NNPDF/eko/blob/01efee1d27d215e895dd055debff8905af3d548c/src/eko/interpolation.py#L439) in line 50

Note that linear here is fine, since we're faking in any case with 3 points in the linear regime. So to recover make `xg` linear, I guess ..

File: tests/yadism/test_tmc.py
Comment (felixhekhorn) [inline tests/yadism/test_tmc.py:50]: this is the `log=False` I meant [here](https://github.com/NNPDF/yadism/pull/157/files#r1013810070)

File: tests/yadism/test_tmc.py
Comment (AleCandido) [reply tests/yadism/test_tmc.py:162]: You're right. Then, I'd try to revert and check if the problem persists.
Can you do this? @andreab1997 @giacomomagni

File: src/yadism/runner.py
Comment (felixhekhorn) [inline src/yadism/runner.py:98]: Actually, even here: you should read the "logginess" from the run card ...

**Review by felixhekhorn — CHANGES_REQUESTED** [review]

File: src/yadism/runner.py
Comment (AleCandido) [reply src/yadism/runner.py:98]: Standard key has been `"interpolation_is_log"` until now. We need a rework of the runcard also for yadism, but not now

File: tests/yadism/test_tmc.py
Comment (giacomomagni) [reply tests/yadism/test_tmc.py]: Thanks @felixhekhorn for the suggestion, indeed this solved the problem!!  :)

File: src/yadism/runner.py
Comment (giacomomagni) [reply src/yadism/runner.py]: I should have solved it!

File: tests/yadism/esf/test_conv.py
Comment (felixhekhorn) [inline tests/yadism/esf/test_conv.py:122]: Apparently it doesn't matter, but still let's keep the old behaviour: so please also change here to linear

File: tests/yadism/test_structure_function.py
Comment (felixhekhorn) [inline tests/yadism/test_structure_function.py:22]: also please here back to linear

**Review by felixhekhorn — APPROVED** [review]

**Pull request merged**
